chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+package co.chatsdk.ui;
+
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
+    }
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+
+        private WeakReference<ImageListAdapter> adapter;
+        public ImageView imageView;
+        public Uri thisImageUri;
+        public int thisImagePosition;
+
+        public MyViewHolder(ImageListAdapter la, ImageView v) {
+            super(v);
+            this.adapter = new WeakReference<ImageListAdapter>(la);
+            imageView = v;
+
+            imageView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    setMainImageDisplay(thisImageUri);
+                    setMainImageUri(thisImageUri);
+                }
+            });
+
+            imageView.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View v) {
+                    Uri toBeDeletedUri = thisImageUri;
+                    removeImageFromAdapter(thisImageUri);
+                    notifyDataSetChanged();
+                    if (getMainImageUri() == toBeDeletedUri) {
```

Comment:
Use `uri.equals()` rather than `===`. That's because `===` will compare the object ID wherease `equals` will compare the URI string. This is important. For example:
```
String s1 = "hello";
String s2 = "hello";
// s1 == s2 = false
// s1.equals(s2) = true
```
Make sure you understand this.

Reply:
Ah, OK, I see.